### ISSUE TITLE
Improve Edit functionality

### ISSUE THREAD

**Issue by alvinlim277** [AY2324S1-CS2103T-W13-1/tp#63]
Currently, EditCommand requires a person type tag, and will edit the person at specified index in the currently displayed list. There are a few issues with this:
1. If a user is viewing patients, they can run a command to edit a specialist without having them on screen, leading to undesirable side effects.
2. Since the edit index is based on the displayed filtered list, the person type is somewhat irrelevant.
3. However, without the personType the current implementation of EditCommandParser does not have an easy way of knowing whether to parse the arguments as a patient editor or specialist editor.

In short more consideration has to be done to either change the syntax or implementation of this command.
